### PR TITLE
[is-odd] Add is-odd typing

### PR DESCRIPTION
**Pull request by azerella**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/is-odd/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Type definitions for is-odd 3.0
+// Project: https://github.com/jonschlinkert/is-odd
+// Definitions by:  Adam Zerella <https://github.com/adamzerella>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ * Return true if a given number is odd or not.
+ */
+declare function isOdd(value: number): boolean;
```
Comment (amcasey) [inline types/is-odd/index.d.ts]: From the sample usage, it looks like the library also accepts strings (and anything `Math.Abs` will convert to a number?).

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/is-odd/index.d.ts (hunk 1 above)
Comment (adamzerella) [reply types/is-odd/index.d.ts]: @amcasey I wasn't sure how to handle something like this. The function would do its best to convert other types even if it wasn't ideal.

`Math.abs()` handles arrays, objects, strings, null etc... Would it be acceptable to change the definition to something like:

`declare function isOdd(value: number|number[]|string|object|undefined): boolean;`

Comment (amcasey) [reply types/is-odd/index.d.ts]: It's a bit of a judgement call.  Personally, I'd support the types shown in the sample usage: `number | string`.

**Comment by adamzerella** [comment]
@amcasey Updated type and added new test 👍

**Review by amcasey — APPROVED** [review]

**Pull request merged**